src/Tizen.BatteryMonitor/Tizen.BatteryMonitor/BatteryMonitor.cs

```diff
@@ -0,0 +1,206 @@
+﻿/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace Tizen.BatteryMonitor
+{
+    static internal class Globals
+    {
+        internal const string LogTag = "Tizen.BatteryMonitor";
+    }
+
+    /// <summary>
+    /// Provides information related to the power consumption by applications or by hardware resources on a battery-powered device for a given duration of time.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public sealed class BatteryMonitor : IDisposable
+    {
+        private static volatile BatteryMonitor _instance = null;
+        private static readonly object _syncLock = new object();
+        private bool _disposed = false;
+
+        /// <summary>
+        /// The constructor of the BatteryMonitor class.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <feature>http://tizen.org/feature/battery</feature>
+        /// <privilege>http://tizen.org/privilege/systemmonitor</privilege>
+        private BatteryMonitor()
+        {
+        }
+
+        /// <summary>
+        /// Returns an instance of BatteryMonitor class
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public static BatteryMonitor Instance
```

Comment:
How about make as `static class`?

Reply:
static class will support all the api's, but considering the flexibility  to refactor the code in future improvements, we thought it would be better if  it is more object oriented, that is why we made it singleton.

Reply:
TizenFX is not use singleton pattern in API, 
You can get flexibility in internal implemenation, it is free, but API should be following consistancy with others APIs

Reply:
If singleton pattern does not provide consistency with other API's we can go with static class